File: C++/Remove every kth node.cpp
class Solution {
    public:
    Node* deleteK(Node *head,int K){
      if(K==1){
          return NULL;
      }
      Node* temp=head;
      Node* pre=NULL;
      long long count=1;
      while(temp!=NULL){
          if(count==K){
              pre->next=temp->next;
              count=0;
          }
          else{
              pre=temp;
          }
          temp=temp->next;
          count++;
      }
      return head;
    }
};

File: JAVA/Remove every kth node.java
class Solution
{
 Node delete(Node head, int k)
    {
    Node temp = head;
    int i=1;
    if(i == k) {
        return null;
    }
    while(temp != null && temp.next != null) {
        if(i+1 == k) {
            temp.next = temp.next.next;
            i = 0;
        }
        i++;
        temp = temp.next;
    }
    return head;
    
}
}
